sdmetrics/reports/multi_table/base_multi_table_report.py

```diff
@@ -32,13 +32,29 @@ def _validate_data_format(self, real_data, synthetic_data):
                 return
 
         error_message = (
-            f'Multi table report {self.__class__.__name__} expects real and synthetic data to be'
+            f'Multi table {self.__class__.__name__} expects real and synthetic data to be'
             ' dictionaries of pandas.DataFrame. If your real and synthetic data are pd.DataFrame,'
             f' please use the single-table {self.__class__.__name__} instead.'
         )
 
         raise ValueError(error_message)
 
+    def _validate_metadata_format(self, metadata):
+        """Validate the metadata."""
+        if not isinstance(metadata, dict):
+            raise TypeError('The provided metadata is not a dictionary.')
+
+        if 'tables' not in metadata:
+            raise ValueError(
+                'Multi table reports expect metadata to contain a "tables" key with a mapping'
+                ' from table names to metadata for each table.'
+            )
+        for table_name, table_metadata in metadata['tables'].items():
+            if 'columns' not in table_metadata:
+                raise ValueError(
+                    f'The metadata for table "{table_name}" is missing a "columns" key.'
+                )
```

Comment:
@npatki should we error in the case where the metadata is malformed? More specifically, f a users gives metadata with an empty table or empty column, should we error?

Reply:
@amontanez24 yeah that's probably a good idea. Though I'm neutral on whether or not there's an explicit error message for this case vs. the report just naturally crashing because it cannot access those key names.